sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/visibility/VisibilityChangeListener.groovy

```diff
@@ -0,0 +1,16 @@
+package life.qbic.portal.sampletracking.components.projectoverview.visibility
+/**
+ * <b>VisibilityChangeListener to handle changes in component visibility</b>
+ *
+ * @since 1.0.0
+ */
+@FunctionalInterface
+interface VisibilityChangeListener {
+
+    /**
+     * This event is fired when the visibility of a component is changed
+     * @param newValue A boolean describing the new visibility value
+     * @param oldValue A boolean describing the old visibility value
+     */
+    void visibilityChangeEvent(boolean newValue, boolean oldValue)
```

Comment:
The common naming for listener methods would be something like `visisbilityChanged` and would expect an Event. I think it makes sense to keep to the standard way of implementing the Listener pattern and to introduce a VisibilityChangeEvent here that is consumed by the `visisbilityChanged` method of the listener.
As we are doing basically the same as the java swing UI listeners (we are handling view events) I would suggest staying with their naming pattern to keep it understandable and unified across the libraries.
https://docs.oracle.com/javase/tutorial/uiswing/events/intro.html